scripts/table_updates/update_table_schema.py

```diff
@@ -19,185 +16,324 @@ def copy_table_schema(syn, from_table_id, to_table_id):
-    irr_table_view = irr_table_view[['id','name']]
-    irr_table_view['name'] = irr_table_view['name'].apply(lambda x: x.replace(' - double curated',''))
-    master_table_view = pandas.merge(sage_table_view, 
-                                     pandas.merge(bpc_table_view,irr_table_view,
-                                                  on='name',suffixes=['_bpc','_irr']),
-                                     on='name')
+    curated_data_element = download_synapse_table(
+        syn, TABLE_INFO["CATALOG_ID"], "dataType='curated'"
+    )
+    curated_data_element = curated_data_element[
+        [
+            "variable",
+            "instrument",
+            "type",
+            "synColType",
+            "synColSize",
+            "numCols",
+            "colLabels",
+        ]
+    ]
+    # create the master table
+    sage_table_view = download_synapse_table(
+        syn, TABLE_INFO["sage"][0], TABLE_INFO["sage"][1]
+    )
+    sage_table_view.drop(columns="table_type", axis=1, inplace=True)
+    bpc_table_view = download_synapse_table(
+        syn, TABLE_INFO["bpc"][0], TABLE_INFO["bpc"][1]
+    )
+    bpc_table_view = bpc_table_view[["id", "name"]]
+    irr_table_view = download_synapse_table(
+        syn, TABLE_INFO["irr"][0], TABLE_INFO["irr"][1]
+    )
+    irr_table_view = irr_table_view[["id", "name"]]
+    irr_table_view["name"] = irr_table_view["name"].apply(
+        lambda x: x.replace(" - double curated", "")
+    )
+    master_table_view = pandas.merge(
+        sage_table_view,
+        pandas.merge(
+            bpc_table_view, irr_table_view, on="name", suffixes=["_bpc", "_irr"]
+        ),
+        on="name",
+    )
     # update table schema for Sage Internal tables
-    form_groups = master_table_view.groupby('form')
+    form_groups = master_table_view.groupby("form")
     for form in form_groups:
         _update_table_schema(syn, form, curated_data_element, logger, dry_run)
     # copy the table schema to update the BPC Internal and IRR tables
     if not dry_run:
         logger.info("Updating table schemas for BPC and IRR tables")
         for _, row in master_table_view.iterrows():
-            new_bpc_schema = copy_table_schema(syn,row['id'],row['id_bpc'])
+            new_bpc_schema = copy_table_schema(syn, row["id"], row["id_bpc"])
             syn.store(new_bpc_schema)
-            new_irr_schema = copy_table_schema(syn,row['id'],row['id_irr'])
+            new_irr_schema = copy_table_schema(syn, row["id"], row["id_irr"])
             syn.store(new_irr_schema)
 
+
 def main():
     parser = argparse.ArgumentParser(
-        description='Update table schema on Synapse Tables for BPC')
+        description="Update table schema on Synapse Tables for BPC"
+    )
     parser.add_argument(
-        "-c", "--synapse_config",
+        "-c",
+        "--synapse_config",
         default=synapseclient.client.CONFIG_FILE,
-        help="Synapse credentials file")
+        help="Synapse credentials file",
```

Comment:
Future improvement but not necessary right now: We can remove anything synapseconfig related because we now use the SYNAPSE_AUTH_TOKEN env variable to pass in credentials.

Reply:
That sounds good to me